src/platform/mbed/net/src/net_socket.cpp

```diff
@@ -1,8 +1,70 @@
 #include <net_socket.h>
 
+BSDSocket sockets[MAX_SOCKET];
+
+Socket * getSocket(int id)
+{
+    int type = 1;
+    switch (type)
+    {
+    case TCP_SOCKET:
+        return &sockets[id].tcpSocket;
+    case UDP_SOCKET:
+        return &sockets[id].udpSocket;
+    default:
+        return nullptr;
+    }
+}
+
+int close(int id)
+{
+    auto * sock = getSocket(id);
+    sock->~Socket();
+    sockets[id].type = SOCKET_NOT_INITIALIZED;
+}
+
 int mbed_socket(int family, int type, int proto)
 {
-    return 0;
+    int id = ERR_NO_MEMORY;
+    for (int i = 0; i < MAX_SOCKET; i++)
+    {
+        if (sockets[i].type == SOCKET_NOT_INITIALIZED)
+        {
+            id = i;
+            break;
+        }
+    }
+    if (id != ERR_NO_MEMORY)
```

Comment:
Would it be possible to return early so it makes the default path clearer and keeps indentation in check: 

```C++
if (id == ERR_NO_MEMORY) { 
    return id; 
}

// process with the switch.

```